python/HeaderService/hslib.py

```diff
@@ -359,23 +360,49 @@ def collect_from_HeaderService(self):
         Collect and update custom meta-data generated or transformed by
         the HeaderService
         """
-        self.DATE_OBS = hutils.get_date_utc(self.metadata['DATE-OBS'])
-        self.DATE_BEG = hutils.get_date_utc(self.metadata['DATE-BEG'])
-        self.DATE_END = hutils.get_date_utc(self.metadata['DATE-END'])
+
+        # Reformat and calculate dates based on different timeStamps
+        self.DATE = hscalc.get_date_utc(self.metadata['DATE'])
+        self.DATE_OBS = hscalc.get_date_utc(self.metadata['DATE-OBS'])
+        self.DATE_BEG = hscalc.get_date_utc(self.metadata['DATE-BEG'])
+        self.DATE_END = hscalc.get_date_utc(self.metadata['DATE-END'])
+        self.metadata['DATE'] = self.DATE.isot
         self.metadata['DATE-OBS'] = self.DATE_OBS.isot
         self.metadata['DATE-BEG'] = self.DATE_BEG.isot
         self.metadata['DATE-END'] = self.DATE_END.isot
+        self.metadata['MJD'] = self.DATE.mjd
         self.metadata['MJD-OBS'] = self.DATE_OBS.mjd
         self.metadata['MJD-BEG'] = self.DATE_BEG.mjd
         self.metadata['MJD-END'] = self.DATE_END.mjd
-        self.metadata['DATE'] = self.DATE_HDR.isot
-        self.metadata['MJD'] = self.DATE_HDR.mjd
+
         self.metadata['FILENAME'] = self.filename_FITS
         # THIS IS AN UGLY HACK TO MAKE IT WORK SEQNUM
         # FIX THIS -- FELIPE, TIAGO, MICHAEL and TIM J. are accomplices
         self.metadata['SEQNUM'] = int(self.metadata['OBSID'].split('_')[-1])
         self.metadata['DAYOBS'] = self.metadata['OBSID'].split('_')[2]
 
+        # The EL/AZ at start
+        if 'ELSTART' and 'AZSTART' in self.metadata:
+            LOGGER.info("Computing RA/DEC from ELSTART/AZSTART")
+            ra, dec = hscalc.get_radec_from_altaz(alt=self.metadata['ELSTART'],
```

Comment:
this is relying on your default position in the function rather than the top-level value that should be passed. I know it's annoying but defaults are asking for trouble later on..

Reply:
Now the top value, as defined by the header template is passed instead.